libvast/src/system/index.cpp

```diff
@@ -771,6 +788,107 @@ index_state::status(status_verbosity v) const {
   return rs->promise;
 }
 
+void schedule_lookups(index_state& st) {
+  if (st.pending_queries.partitions.empty())
+    return;
+  auto t = timer::start(st.scheduler_measurement);
+  auto num_scheduled = size_t{0};
+  auto on_return = caf::detail::make_scope_guard([&] {
+    t.stop(num_scheduled);
+  });
+  while (st.running_partition_lookups < st.max_concurrent_partition_lookups) {
+    // 1. Get the partition with the highest accumulated priority.
```

Comment:
Before this PR we maintain a backlog with queries that weren't sent to the `catalog` yet. After this change we do the catalog lookup immediately. The purpose of this change is to avoid "unloading" partitions from memory when we know they will be needed for other queries as well. Maintaining completely separate queues doesn't make sense for this scheme, so we now use the priority as a scoring factor when sorting the queue.

Reply:
TODO: We decide to make it possible to switch to the old behavior with a config option.

Reply:
This has been implemented in 8beee922e94d50fd41384575ad925bc9a34e2eeb.